core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -133,9 +143,14 @@ trait MethodConfigurationComponent {
       loadMethodConfigurationById(methodConfigurationId)
     }
 
-    def rename(workspaceContext: SlickWorkspaceContext, methodConfigurationNamespace: String, methodConfigurationName: String, newName: String): ReadWriteAction[Int] = {
-      workspaceQuery.updateLastModified(workspaceContext.workspaceId) andThen
-        findByName(workspaceContext.workspaceId, methodConfigurationNamespace, methodConfigurationName).map(_.name).update(newName)
+    // To rename a method config, we "delete" (or rather hide) the old version of the method config and create a new row for the new method config, with the
+    //  config version # incremented
+    def rename(workspaceContext: SlickWorkspaceContext, methodConfigurationNamespace: String, methodConfigurationName: String, newMethodConfig: MethodConfiguration) = {
+      uniqueResult[MethodConfigurationRecord](findByName(workspaceContext.workspaceId, methodConfigurationNamespace, methodConfigurationName)) flatMap {
+        case None => DBIO.successful(false)
```

Comment:
The issue I had  with getting the method config record in WorkspaceService is that I'm assuming don't use MethodConfigurationRecord outside of MethodConfigurationComponent, in which case maybe it makes more sense to  find the record here and change the method that calls rename in WorkspaceService so it doesn't have withMethodConfig?